reana_server/validation.py

```diff
@@ -125,3 +134,20 @@ def validate_workflow(reana_yaml: Dict, input_parameters: Dict) -> None:
     validate_compute_backends(reana_yaml)
     validate_workspace_path(reana_yaml)
     validate_inputs(reana_yaml)
+
+
+def validate_retention_rule(rule: str) -> None:
+    """Validate retention rule.
+
+    :param rule: retention rule
+    :type rule: string
+
+    :raises reana_commons.errors.REANAValidationError: if rule is not valid
+    """
+    rule_path = pathlib.Path(rule)
+    if rule_path.is_absolute():
+        raise REANAValidationError(f"Retention rule {rule} cannot be an absolute path")
+    if not rule_path.parts:
+        raise REANAValidationError(f"Retention rule {rule} cannot be empty")
+    if ".." in rule_path:
+        raise REANAValidationError(f"Retention rule {rule} cannot contain '..'")
```

Comment:
This part is similar to `validate_inputs`. Do you think extracting this block to a common function like `validate_local_path` makes sense?

Reply:
I personally think that it does not make too much sense to generalize this, given that in one case we are validating paths, and in the other we are validating patterns with wildcards